# RCS1236: Use exception filter

| Property                 | Value   |
| ------------------------ | ------- |
| Id                       | RCS1236 |
| Category                 | Usage   |
| Severity                 | Info    |
| Minimal Language Version | 6\.0    |

## Example

### Code with Diagnostic

```csharp
try
{
}
catch (Exception ex)
{
    if (!(ex is InvalidOperationException)) // RCS1236
    {
        throw;
    }

    return;
}
```

### Code with Fix

```csharp
try
{
}
catch (Exception ex) when (ex is InvalidOperationException)
{
    return;
}
```

## Remarks

The accepted [answer](https://stackoverflow.com/a/28879616) from Stack Overflow states:
>"If there is an exception thrown within the filter, then that exception will be silently swallowed and the filter simply fails."

It is impossible to definitely detect if an exception can be thrown within an expression.

So the expression is considered as the one that can throw an exception if it contains method call that meets one of the following requirements:

* Its name starts with `ThrowIf`.
* Its XML comment contains `<exception>` element.

## See Also

* [What's New in C# 6.0 - Exception Filters](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-6#exception-filters)
* [Exception filters in C# 6: their biggest advantage is not what you think](https://thomaslevesque.com/2015/06/21/exception-filters-in-c-6/)
* [How to Suppress a Diagnostic](../HowToConfigureAnalyzers.md#how-to-suppress-a-diagnostic)


*\(Generated with [DotMarkdown](http://github.com/JosefPihrt/DotMarkdown)\)*